IPNI_FEDERATION.md

```diff
@@ -0,0 +1,406 @@
+multiple IPNI instances but as a robust mechanism to ensure that the distributed nature of the system did not compromise
+on the uniformity and reliability of content discovery. This document delves deep into this protocol, exploring its
+intricacies and mechanisms, laying the foundation for a truly inter-planetary scale network indexer.
+
+## Motivation
+
+The underlying motivation behind enhancing IPNI is manifold. Foremost, guaranteeing an eventually consistent view of
+index records across multiple IPNI instances holds the potential for transformative benefits. Such consistency paves the
+way for reducing trust in an individual indexer.
+
+## Non-goals
+
+This section clarifies aspects that the IPNI federation protocol will not address in its current iteration.
+Here are the non-goals for the IPNI federation protocol:
+
+* **Permissionless Membership**: This specification assumes a permissioned network, where the membership is controlled
+  and not open to arbitrary participants.
+* **Real-time synchronization**: The protocol strives for eventual consistency, but real-time synchronization across all
+  nodes, especially in a globally distributed setup, is not the objective.
+* **Network Topology Design**: The IPNI federation protocol does not define or restrict the underlying network topology.
+  Whether nodes are structured in a mesh, star, or hierarchical format, the protocol remains agnostic.
+* **Strong consistency**: While the IPNI federation protocol aims to ensure data consistency and availability, it does
+  not aim to atomically replicate every piece of indexed data across all nodes. For example, it is possible for some
+  records to remain partially available on one node until deletions are propagate through the network.
+
+## Terminology
+
+* **Epoch**: An epoch in the context of the IPNI federation protocol refers to a specific time period or iteration of
+  the protocol's operation. It can be thought of as a version or generation number that increments each time there's a
+  major operation or update, aiding in synchronization efforts across instances.
+
+* **Reconciliation**: Reconciliation is the process of ensuring that two or more IPNI instances have consistent index
+  records. It involves comparing the content of instances, identifying discrepancies, and making updates so that all
+  participating instances have the same content.
+
+* **Vector Clock (VC)**: Vector Clock is a mechanism used to determine the order of events in a distributed system. In
+  IPNI federation protocol, a vector clock helps track the sequence of index updates across multiple instances, thereby
+  assisting in maintaining the eventual consistency of the index.
+
+* **Snapshot**: A snapshot refers to a captured state of the index records at a particular point in time consisting of
+  the providers list known by the indexer, the latest processed advertisement CID for each provider and an epoch number.
```

Comment:
I think it would be useful to also have a link to the previous epoch. This would allow walking a chain by epoch, instead of having to iterate through each advertisement, if something deep in the chain needed to be accessed.

Reply:
It does also have link to previous. I'll update the description here 👍